dask/bytes/core.py

```diff
@@ -390,7 +390,7 @@ def get_fs_paths_myopen(urlpath, compression, mode, encoding='utf8',
                                  encoding=encoding, errors=errors, **kwargs)
         if 'w' in mode:
             paths = _expand_paths(urlpath, name_function, num)
-        elif "*" in urlpath:
+        elif "*" in urlpath or "?" in urlpath or "[" in urlpath:
```

Comment:
According to [docs](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html), one can use `*` in S3 keys, `?` is also possible, but requires special handling, and `[]` are the characters to avoid.